src/data/hooks/useUpdateRecipe.ts

```diff
@@ -0,0 +1,57 @@
+import { useMutation } from "@apollo/client";
+import { gql } from "__generated__";
+import { IngredientRefInfo } from "__generated__/graphql";
+import { toMilliseconds } from "util/time";
+import { DraftRecipe } from "global/types/types";
+
+const UPDATE_RECIPE_MUTATION = gql(`
+mutation updateRecipe($id: ID!, $info: IngredientInfo!, $photo: Upload) {
+  library {
+    updateRecipe(id: $id, info: $info, photo: $photo) {
+      id
+      name
+    }
+  }
+}
+`);
+
+export const useUpdateRecipe = () => {
+    const [mutateFunction, { data, loading, error }] = useMutation(
+        UPDATE_RECIPE_MUTATION,
+    );
+
+    const updateRecipe = (recipe: DraftRecipe) => {
+        return mutateFunction({
+            variables: {
+                id: recipe.id.toString(),
+                info: {
```

Comment:
This `DraftRecipe` -> `IngredientInfo` repackaging is identical between create/update. Worth abstracting into a helper?

Reply:
Oooh I somehow missed this comment. Good idea.